tardis/interfaces/siteadapter.py

```diff
@@ -13,17 +13,27 @@ class ResourceStatus(Enum):
 
 
 class SiteAdapter(metaclass=ABCMeta):
+    @property
+    def configuration(self) -> AttributeDict:
+        try:
+            # noinspection PyUnresolvedReferences
+            return self._configuration
+        except AttributeError as ae:
+            raise AttributeError(
+                f"Class {self.__class__.__name__} must have an '_configuration' instance variable"  # noqa
+            ) from ae
+
     @abstractmethod
     async def deploy_resource(
         self, resource_attributes: AttributeDict
     ) -> AttributeDict:
-        return NotImplemented
+        raise NotImplementedError
 
     def drone_uuid(self, uuid) -> str:
         return f"{self.site_name.lower()}-{uuid}"
 
     def handle_exceptions(self):
-        return NotImplemented
+        raise NotImplementedError
```

Comment:
What do you think about making this also an ``@abstractmethod``?

Reply:
Good catch! I will change it.